File: backend/tests/accounts/factories.py
from accounts.models import Account, Transaction
from users.models import User


def create_account(owner: User, **overrides: object) -> Account:
    defaults: dict[str, object] = {"name": "Amazon", "type": Account.Type.GIFT_CARD}
    defaults.update(overrides)
    return Account.objects.create(user=owner, **defaults)


def create_transaction(account: Account, **overrides: object) -> Transaction:
    defaults: dict[str, object] = {"amount": "10.00", "occurred_on": "2026-01-01"}
    defaults.update(overrides)
    return Transaction.objects.create(account=account, **defaults)

File: web/tests/features/accounts/factories.ts
import type { Account } from "@/features/accounts/types";

export function makeAccount(overrides: Partial<Account> = {}): Account {
  return {
    id: "1",
    name: "Amazon",
    description: "",
    type: "gift_card",
    expires_on: null,
    created_at: "2026-01-01T00:00:00Z",
    updated_at: "2026-01-01T00:00:00Z",
    ...overrides,
  };
}
